### ISSUE TITLE
NPE when parsing AWS price file

### ISSUE THREAD

**Issue by thaussma** [nextflow-io/nextflow#408]
The `nextflow cloud` command is failing (all of a sudden?) to fetch the AWS spot prices and cannot create AWS clusters anymore.

```bash
$ nextflow cloud spot-prices
ERROR ~ Cannot cast object 'null' with class 'null' to class 'int'. Try 'java.lang.Integer' instead
```

```bash
$ nextflow cloud create test
> cluster name: test
> instances count: 1
> Launch configuration:
 - driver: 'aws'
 - imageId: 'ami-XXXXX'
 - instanceType: 'm4.large'
 - keyFile: /Users/XXX
 - sharedStorageId: 'fs-XXXXXX'
 - sharedStorageMount: '/mnt/shared'
 - subnetId: 'subnet-XXXXX'
 - userName: 'XXXXXX'

Please confirm you really want to launch the cluster with above configuration [y/n] y
ERROR ~ Cannot cast object 'null' with class 'null' to class 'int'. Try 'java.lang.Integer' instead
```

I own a Mac with El Captain (10.11.6)
```bash
$ nextflow -version

      N E X T F L O W
      Version 0.25.2 build 4462
      last modified 03-07-2017 08:47 UTC (10:47 CEST)
      cite doi:10.1038/nbt.3820
      http://nextflow.io

$ java -version
java version "1.8.0_45"
Java(TM) SE Runtime Environment (build 1.8.0_45-b14)
Java HotSpot(TM) 64-Bit Server VM (build 25.45-b02, mixed mode)

$ groovy -version
Groovy Version: 2.4.7 JVM: 1.8.0_45 Vendor: Oracle Corporation OS: Mac OS X
```

**Comment by pditommaso** [comment]
Please re-execute the command as shown below and attach the create `log` file 

```
nextflow -log log cloud create test
```

**Comment by thaussma** [comment]
I used `brew upgrade nextflow` and `nextflow self-update` and I do not remember how I installed `nextflow` in the first place.

```bash
$ cat log
Jul-25 12:05:59.719 [main] DEBUG nextflow.cli.Launcher - $> /usr/local/bin/nextflow -log log cloud create test
Jul-25 12:05:59.767 [main] DEBUG nextflow.cloud.CloudDriverFactory - Discovered cloud driver: `aws` [nextflow.cloud.aws.AmazonCloudDriver]
Jul-25 12:05:59.780 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: nextflow.config
Jul-25 12:05:59.781 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/xxx/private/next-test/nextflow.config
Jul-25 12:05:59.994 [main] DEBUG nextflow.config.ConfigBuilder - Setting config profile: 'standard'
Jul-25 12:06:00.134 [main] DEBUG nextflow.Global - Using AWS credential defined in `default` section in file: /Users/xxx/.aws/credentials
Jul-25 12:06:05.457 [main] ERROR nextflow.cli.Launcher - @unknown
org.codehaus.groovy.runtime.typehandling.GroovyCastException: Cannot cast object 'null' with class 'null' to class 'int'. Try 'java.lang.Integer' instead
	at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.castToNumber(DefaultTypeTransformation.java:163)
	at org.codehaus.groovy.runtime.typehandling.DefaultTypeTransformation.intUnbox(DefaultTypeTransformation.java:69)
	at nextflow.cloud.aws.AmazonPriceReader.parse(AmazonPriceReader.groovy:192)
	at nextflow.cloud.aws.AmazonPriceReader.getCachedTable(AmazonPriceReader.groovy:116)
	at nextflow.cloud.aws.AmazonPriceReader.getInstanceTypeTable(AmazonPriceReader.groovy:104)
	at nextflow.cloud.aws.AmazonPriceReader.getInstanceType(AmazonPriceReader.groovy:92)
	at nextflow.cloud.aws.AmazonCloudDriver.memoizedMethodPriv$describeInstanceTypeString(AmazonCloudDriver.groovy:664)
	at nextflow.cloud.aws.AmazonCloudDriver$_closure1.doCall(AmazonCloudDriver.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at groovy.lang.Closure.call(Closure.java:414)
	at org.codehaus.groovy.runtime.memoize.Memoize$MemoizeFunction.call(Memoize.java:134)
	at groovy.lang.Closure.call(Closure.java:430)
	at nextflow.cloud.aws.AmazonCloudDriver.describeInstanceType(AmazonCloudDriver.groovy)
	at nextflow.cloud.aws.AmazonCloudDriver.validate(AmazonCloudDriver.groovy:614)
	at nextflow.cloud.CloudConfig.validate(CloudConfig.groovy:253)
	at nextflow.cloud.CloudConfig$validate$5.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.cli.CmdCloud.addNodes0(CmdCloud.groovy:277)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnCurrentN(ScriptBytecodeAdapter.java:82)
	at nextflow.cli.CmdCloud.this$dist$invoke$2(CmdCloud.groovy)
	at nextflow.cli.CmdCloud$CreateCluster.methodMissing(CmdCloud.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaClassImpl.invokeMissingMethod(MetaClassImpl.java:939)
	at groovy.lang.MetaClassImpl.invokePropertyOrMissing(MetaClassImpl.java:1262)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1215)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:69)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
	at nextflow.cli.CmdCloud$CreateCluster.apply(CmdCloud.groovy:504)
	at nextflow.cli.CmdCloud$SubCmd$apply.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.cli.CmdCloud.run(CmdCloud.groovy:178)
	at nextflow.cli.Launcher.run(Launcher.groovy:410)
	at nextflow.cli.Launcher.main(Launcher.groovy:564)
```

**Comment by pditommaso** [comment]
Thanks for reporting this issue. I've uploaded a new snapshot that should solve this problem. To use it declare in your env environment this variable: 

```
export NXF_VER=0.25.3-SNAPSHOT
```

Then use NF as usual. Please report if this solve the issue.

**Comment by thaussma** [comment]
The snapshot throws a `NPE`:
```
14:18 $ cat log
Jul-25 14:18:21.316 [main] DEBUG nextflow.cli.Launcher - $> /usr/local/bin/nextflow -log log cloud create test
Jul-25 14:18:21.366 [main] DEBUG nextflow.cloud.CloudDriverFactory - Discovered cloud driver: `aws` [nextflow.cloud.aws.AmazonCloudDriver]
Jul-25 14:18:21.379 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: nextflow.config
Jul-25 14:18:21.380 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: /Users/xxx/private/next-test/nextflow.config
Jul-25 14:18:21.597 [main] DEBUG nextflow.config.ConfigBuilder - Setting config profile: 'standard'
Jul-25 14:18:21.732 [main] DEBUG nextflow.Global - Using AWS credential defined in `default` section in file: /Users/xxx/.aws/credentials
Jul-25 14:18:24.904 [main] ERROR nextflow.cli.Launcher - @unknown
java.lang.NullPointerException: null
	at nextflow.cloud.aws.AmazonPriceReader.parseMem(AmazonPriceReader.groovy:242)
	at nextflow.cloud.aws.AmazonPriceReader.parse(AmazonPriceReader.groovy:193)
	at nextflow.cloud.aws.AmazonPriceReader.getCachedTable(AmazonPriceReader.groovy:116)
	at nextflow.cloud.aws.AmazonPriceReader.getInstanceTypeTable(AmazonPriceReader.groovy:104)
	at nextflow.cloud.aws.AmazonPriceReader.getInstanceType(AmazonPriceReader.groovy:92)
	at nextflow.cloud.aws.AmazonCloudDriver.memoizedMethodPriv$describeInstanceTypeString(AmazonCloudDriver.groovy:664)
	at nextflow.cloud.aws.AmazonCloudDriver$_closure1.doCall(AmazonCloudDriver.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at org.codehaus.groovy.runtime.metaclass.ClosureMetaClass.invokeMethod(ClosureMetaClass.java:294)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at groovy.lang.Closure.call(Closure.java:414)
	at org.codehaus.groovy.runtime.memoize.Memoize$MemoizeFunction.call(Memoize.java:134)
	at groovy.lang.Closure.call(Closure.java:430)
	at nextflow.cloud.aws.AmazonCloudDriver.describeInstanceType(AmazonCloudDriver.groovy)
	at nextflow.cloud.aws.AmazonCloudDriver.validate(AmazonCloudDriver.groovy:614)
	at nextflow.cloud.CloudConfig.validate(CloudConfig.groovy:253)
	at nextflow.cloud.CloudConfig$validate$5.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.cli.CmdCloud.addNodes0(CmdCloud.groovy:277)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaMethod.doMethodInvoke(MetaMethod.java:325)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1213)
	at org.codehaus.groovy.runtime.ScriptBytecodeAdapter.invokeMethodOnCurrentN(ScriptBytecodeAdapter.java:82)
	at nextflow.cli.CmdCloud.this$dist$invoke$2(CmdCloud.groovy)
	at nextflow.cli.CmdCloud$CreateCluster.methodMissing(CmdCloud.groovy)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.codehaus.groovy.reflection.CachedMethod.invoke(CachedMethod.java:93)
	at groovy.lang.MetaClassImpl.invokeMissingMethod(MetaClassImpl.java:939)
	at groovy.lang.MetaClassImpl.invokePropertyOrMissing(MetaClassImpl.java:1262)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1215)
	at groovy.lang.MetaClassImpl.invokeMethod(MetaClassImpl.java:1022)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.callCurrent(PogoMetaClassSite.java:69)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCallCurrent(CallSiteArray.java:52)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:154)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:182)
	at nextflow.cli.CmdCloud$CreateCluster.apply(CmdCloud.groovy:504)
	at nextflow.cli.CmdCloud$SubCmd$apply.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at nextflow.cli.CmdCloud.run(CmdCloud.groovy:178)
	at nextflow.cli.Launcher.run(Launcher.groovy:410)
	at nextflow.cli.Launcher.main(Launcher.groovy:564)
```

**Comment by pditommaso** [comment]
What region are you using ?

**Comment by thaussma** [comment]
```
[default]
region = eu-west-1
output = text
```

**Comment by pditommaso** [comment]
OK, I've managed to replicate the issue. This time should be fine. Update your snapshot with the command: 

```
CAPSULE_RESET=1 NXF_VER=0.25.3-SNAPSHOT nextflow info 
```

It should print 

```
  Version: 0.25.3-SNAPSHOT build 4481
  Modified: 25-07-2017 14:00 UTC (16:00 CEST)
```

Then run as before having the variable `NXF_VER=0.25.3-SNAPSHOT` define in your environment.

**Comment by thaussma** [comment]
Thank you for the instant fix. It's working!

**Comment by pditommaso** [comment]
Released with version `0.25.3`.
